src/node_builtins.h

```diff
@@ -25,11 +26,15 @@ class Realm;
 
 namespace builtins {
 
-using BuiltinSourceMap = std::map<std::string, UnionBytes>;
+using InternalizedBuiltinSourceMap = std::map<std::string, EternalBytes>;
```

Comment:
Actually before https://github.com/nodejs/node/pull/46119 UnionBytes never owns the underlying data and is more or less doing what EternalBytes does here, so we could've just register all the references to the static char arrays and made it work. I wonder if we can make it simpler by instead of recreating an EternalBytes class, we create an owning version of the UnionBytes class, and use that for externalized builtins (which is less common because that requires a build-time flag) with the owning ptr, and make UnionBytes owning again.

Reply:
That makes sense.

> so we could've just register all the references to the static char arrays and made it work

To be pedantic here (because it tripped me up on the first iteration of this diff), we need to register references to `v8::String::ExternalStringResource`, not the static char arrays.

> Otherwise we end up with an EternalBytes that's non-owning, and an UnionBytes class that still does both, which is more complicated than having one owning class and one non-owning class.

There actually is not any users of the non-owning version of UnionBytes after this PR, i.e. I can make the constructors private:

```diff
diff --git a/src/node_union_bytes.h b/src/node_union_bytes.h
index 6e1a3afb61..cff5ebc0ff 100644
--- a/src/node_union_bytes.h
+++ b/src/node_union_bytes.h
@@ -16,10 +16,6 @@ namespace node {
 // via ToStringChecked.
 class UnionBytes {
  public:
-  UnionBytes(const uint16_t* data, size_t length)
-      : one_bytes_(nullptr), two_bytes_(data), length_(length) {}
-  UnionBytes(const uint8_t* data, size_t length)
-      : one_bytes_(data), two_bytes_(nullptr), length_(length) {}
   template <typename T>  // T = uint8_t or uint16_t
   explicit UnionBytes(std::shared_ptr<std::vector</*const*/ T>> data)
       : UnionBytes(data->data(), data->size()) {
@@ -44,6 +40,11 @@ class UnionBytes {
   size_t length() const { return length_; }
 
  private:
+  UnionBytes(const uint16_t* data, size_t length)
+      : one_bytes_(nullptr), two_bytes_(data), length_(length) {}
+  UnionBytes(const uint8_t* data, size_t length)
+      : one_bytes_(data), two_bytes_(nullptr), length_(length) {}
+
   const uint8_t* one_bytes_;
   const uint16_t* two_bytes_;
   size_t length_;
```

so you're saying we should rename `UnionBytes -> OwningUnionBytes` and `EternalBytes -> UnionBytes`, correct? I don't feel particularly strongly about the names, so happy to oblige.

Reply:
If the reference point is the main branch then the idea is to add a new `OwningUnionBytes` class. If the reference point is the current state of this PR then yes it'd be more like renaming `UnionBytes -> OwningUnionBytes` and `EternalBytes -> UnionBytes`.

Reply:
Resolving this conversation since there is now only one UnionBytes class again.